cli/internal/context/context.go

```diff
@@ -377,31 +427,40 @@ func GetTargetsFromArguments(arguments []string, configJson *fs.TurboConfigJSON)
 func (c *Context) populateTopologicGraphForPackageJson(pkg *fs.PackageJSON) error {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
+	depMap := make(map[string]string)
```

Comment:
I don't particularly like this, but I need a map to keep track of the versions and can't just use the `dag.Set`. We still "need" depSet because it is used to calculate the `externalUnresolvedDepsSet` by comparing against the internalDepsSet.

```go
externalUnresolvedDepsSet := depSet.Difference(internalDepsSet)
```

I'm guessing this would be better if we just add an else branch below and do:
```go
      if isInternal {
				internalDepsSet.Add(item.Name)
				c.TopologicalGraph.Connect(dag.BasicEdge(pkg.Name, dependencyName))
			} else {
			  externalUnresolvedDepsSet.Add(item.Name)
      }
```

Reply:
Actually the reason I didn't do that is because there are two places you need to update external deps:
```go
	// split out internal vs. external deps
	for dependencyName, version := range depMap {
		if item, ok := c.PackageInfos[dependencyName]; ok {
			isInternal, err := IsReferenceToLocalPackage(item, version)
			if err != nil {
				return err
			}
			if isInternal {
				internalDepsSet.Add(item.Name)
				c.TopologicalGraph.Connect(dag.BasicEdge(pkg.Name, dependencyName))
			} else {
				externalUnresolvedDepsSet.Add(item.Name);
			}
		} else {
			externalUnresolvedDepsSet.Add(item.Name);
		}
	}

```

Reply:
The revised version of this loop is now:
```go
	for dependencyName, version := range depMap {
		isInternal, err := IsReferenceToLocalPackage(c.PackageInfos, dependencyName, version)

		if err != nil {
			return err
		}

		if isInternal {
			internalDepsSet.Add(dependencyName)
			c.TopologicalGraph.Connect(dag.BasicEdge(pkg.Name, dependencyName))
		} else {
			externalUnresolvedDepsSet.Add(dependencyName)
		}
	}
```